File: Hunt_Career_Cypress/cypress/fixtures/employerRegisterData.js
import {
  generateEmail,
  generateRandomPassword,
  getRandomCompanyName
} from "../utils/utils";

export const registerData = {
  validUser:[ 
  (() => {
    const password = generateRandomPassword();
    return {
      description: "Valid User",
      name: getRandomCompanyName(),
      email: generateEmail("normal"),
      password: password,
      confirmPassword: password,
      expectedSuccess: "checkSuccessCreationMessage"
    };
  })(),

  (() => {
    const password = generateRandomPassword();
    return {
      description: "Email With Dot InBetween Name",
      name: getRandomCompanyName(),
      email: generateEmail("dotBetweenName"),
      password: password,
      confirmPassword: password,
      expectedSuccess: "checkSuccessCreationMessage"
  }
})(),

(() => {
    const password = generateRandomPassword();
    return {
    description: "Sapce Before Name",
    name: getRandomCompanyName(),
      email: generateEmail("spaceBeforeName"),
      password: password,
      confirmPassword: password,
      expectedSuccess: "checkSuccessCreationMessage"
  }
})(),

(() => {
    const password = generateRandomPassword();
    return {
    description: "Space After Name",
    name: getRandomCompanyName(),
      email: generateEmail("spaceAfterName"),
      password: password,
      confirmPassword: password,
      expectedSuccess: "checkSuccessCreationMessage"
  }
})(),
],

  duplicateUser: {
    description: "Duplicate employer",
    name: "Exigent Services Pvt Ltd",
    email: "exigent@mailinator.com",
    password: "Test@1234",
    confirmPassword: "Test@1234",
    expectedError: "checkDuplicateErrorMessage"
  },

  invalidCases: [
    {
      description: "Missing Name",
      name: "",
      email: generateEmail("normal"),
      password: "Test@1234",
      confirmPassword: "Test@1234",
      expectedError: "checkEmployerNameBlankMessage",
    },
    {
      description: "Missing Email",
      name: getRandomCompanyName(),
      email: "",
      password: "Test@1234",
      confirmPassword: "Test@1234",
      expectedError: "checkEmailBlankMessage",
    },
    {
      description: "Invalid Email Format (Plain)",
      name: getRandomCompanyName(),
      email: generateEmail("plainEmail"),
      password: "Test@1234",
      confirmPassword: "Test@1234",
      expectedError: "checkPleaseEnterValidEmailAddressMessage",
    },
    {
      description: "Invalid Email Format (No Local Part)",
      name: getRandomCompanyName(),
      email: generateEmail("noLocalPart"),
      password: "Test@1234",
      confirmPassword: "Test@1234",
      expectedError: "checkPleaseEnterValidEmailAddressMessage",
    },
    {
      description: "Invalid Email Format (Local Part Only)",
      name: getRandomCompanyName(),
      email: generateEmail("localPartOnly"),
      password: "Test@1234",
      confirmPassword: "Test@1234",
      expectedError: "checkPleaseEnterValidEmailAddressMessage",
    },
    {
      description: "Invalid Email Format (Two @)",
      name: getRandomCompanyName(),
      email: generateEmail("twoConsecutiveAt"),
      password: "Test@1234",
      confirmPassword: "Test@1234",
      expectedError: "checkPleaseEnterValidEmailAddressMessage",
    },
    {
      description: "Invalid Email Format (No Top Level Domain)",
      name: getRandomCompanyName(),
      email: generateEmail("withoutTopLevelDomain"),
      password: "Test@1234",
      confirmPassword: "Test@1234",
      expectedError: "checkPleaseEnterValidEmailAddressMessage",
    },
    {
      description: "Invalid Email Format (Trailing Dot)",
      name: getRandomCompanyName(),
      email: generateEmail("tailingWithDot"),
      password: "Test@1234",
      confirmPassword: "Test@1234",
      expectedError: "checkPleaseEnterValidEmailAddressMessage",
    },
    {
      description: "Invalid Email Format (Short TLD)",
      name: getRandomCompanyName(),
      email: generateEmail("topLevelDomainTooShort"),
      password: "Test@1234",
      confirmPassword: "Test@1234",
      expectedError: "checkPleaseEnterValidEmailAddressMessage",
    },
    {
      description: "Invalid Email Format (Domain Starts With Hyphen)",
      name: getRandomCompanyName(),
      email: generateEmail("domainStartsWithHyphen"),
      password: "Test@1234",
      confirmPassword: "Test@1234",
      expectedError: "checkPleaseEnterValidEmailAddressMessage"
    },
    {
      description: "Invalid Email Format (Double Dot in Domain)",
      name: getRandomCompanyName(),
      email: generateEmail("doubleDotInDomain"),
      password: "Test@1234",
      confirmPassword: "Test@1234",
      expectedError: "checkPleaseEnterValidEmailAddressMessage"
    },
    {
      description: "Invalid Email Format (No @)",
      name: getRandomCompanyName(),
      email: generateEmail("noAtTheRate"),
      password: "Test@1234",
      confirmPassword: "Test@1234",
      expectedError: "checkPleaseEnterValidEmailAddressMessage",
    },
    {
      description: "Invalid Email Format (Illegal Characters)",
      name: getRandomCompanyName(),
      email: generateEmail("illegalCharacters"),
      password: "Test@1234",
      confirmPassword: "Test@1234",
      expectedError: "checkPleaseEnterValidEmailAddressMessage",
    },
    {
      description: "Invalid Email Format (Dot After Name)",
      name: getRandomCompanyName(),
      email: generateEmail("dotAfterName"),
      password: "Test@1234",
      confirmPassword: "Test@1234",
      expectedError: "checkPleaseEnterValidEmailAddressMessage",
    },
    {
      description: "Invalid Email Format (Two Consecutive Dots)",
      name: getRandomCompanyName(),
      email: generateEmail("twoConsecutiveDots"),
      password: "Test@1234",
      confirmPassword: "Test@1234",
      expectedError: "checkPleaseEnterValidEmailAddressMessage",
    },
    {
      description: "Invalid Email Format (Dot At Beginning)",
      name: getRandomCompanyName(),
      email: generateEmail("dotAtBeginning"),
      password: "Test@1234",
      confirmPassword: "Test@1234",
      expectedError: "checkPleaseEnterValidEmailAddressMessage",
    },
    {
      description: "Invalid Email Format (With Quotation)",
      name: getRandomCompanyName(),
      email: generateEmail("withQuotation"),
      password: "Test@1234",
      confirmPassword: "Test@1234",
      expectedError: "checkPleaseEnterValidEmailAddressMessage",
    },
    {
      description: "Invalid Email Format (Unclosed Quotation)",
      name: getRandomCompanyName(),
      email: generateEmail("unclosedQuotation"),
      password: "Test@1234",
      confirmPassword: "Test@1234",
      expectedError: "checkPleaseEnterValidEmailAddressMessage",
    },
    {
      description: "Invalid Email Format (Very Lengthy Local Part)",
      name: getRandomCompanyName(),
      email: generateEmail("veryLengthyLocal"),
      password: "Test@1234",
      confirmPassword: "Test@1234",
      expectedError: "checkPleaseEnterValidEmailAddressMessage",
    },
    {
      description: "Invalid Email Format (Very Lengthy Domain)",
      name: getRandomCompanyName(),
      email: generateEmail("veryLengthyDomain"),
      password: "Test@1234",
      confirmPassword: "Test@1234",
      expectedError: "checkPleaseEnterValidEmailAddressMessage",
    },
    {
      description: "Invalid Email Format (Very Lengthy TLD)",
      name: getRandomCompanyName(),
      email: generateEmail("veryLengthyTopLevelDomain"),
      password: "Test@1234",
      confirmPassword: "Test@1234",
      expectedError: "checkPleaseEnterValidEmailAddressMessage",
    },
    {
      description: "Invalid Email Format (Missing Second Level Domain)",
      name: getRandomCompanyName(),
      email: generateEmail("missingSecondLevelDomain"),
      password: "Test@1234",
      confirmPassword: "Test@1234",
      expectedError: "checkPleaseEnterValidEmailAddressMessage",
    },
    {
      description: "Invalid Email Format (With Comma)",
      name: getRandomCompanyName(),
      email: generateEmail("withComma"),
      password: "Test@1234",
      confirmPassword: "Test@1234",
      expectedError: "checkPleaseEnterValidEmailAddressMessage",
    },
    {
      description: "Invalid Email Format (Underscore in TLD)",
      name: getRandomCompanyName(),
      email: generateEmail("underscoreInTLD"),
      password: "Test@1234",
      confirmPassword: "Test@1234",
      expectedError: "checkPleaseEnterValidEmailAddressMessage",
    },
    {
      description: "Invalid Email Format (Without Domain)",
      name: getRandomCompanyName(),
      email: generateEmail("withoutDomain"),
      password: "Test@1234",
      confirmPassword: "Test@1234",
      expectedError: "checkPleaseEnterValidEmailAddressMessage",
    },
    {
      description: "Short Password",
      name: getRandomCompanyName(),
      email: generateEmail("normal"),
      password: "Test",
      confirmPassword: "Test",
      expectedError: "checkPasswordMustHaveRequiredNumberCharactersMessage"
    },
    {
      description: "Password Mismatch",
      name: getRandomCompanyName(),
      email: generateEmail("normal"),
      password: "Test@1234",
      confirmPassword: "Mismatch@1234",
      expectedError: "checkPasswordNotMatchMessage",
    },
    {
      description: "No Password",
      name: getRandomCompanyName(),
      email: generateEmail("normal"),
      password: " ",
      confirmPassword: "Test@1234",
      expectedError: "checkPasswordMustHaveRequiredNumberCharactersMessage",
    },
    {
      description: "No Confirm Password",
      name: getRandomCompanyName(),
      email: generateEmail("normal"),
      password: "Test@1234",
      confirmPassword: " ",
      expectedError: "checkPasswordNotMatchMessage",
    }
  ]
};

File: Hunt_Career_TestNG_Hybrid_Framework/src/main/java/com/huntcareer/qa/testdata/EmployerRegisterData.java
package com.huntcareer.qa.testdata;

import java.util.HashMap;
import java.util.Map;

import com.huntcareer.qa.utils.Utilities;

public class EmployerRegisterData {
    public static Map<String, String> validUser() {
        String password = Utilities.generateRandomPassword();
        Map<String, String> data = new HashMap<>();
        data.put("name", Utilities.getCompanyName());
        data.put("email", Utilities.getEmail("default"));
        data.put("password", password);
        data.put("confirmPassword", password);
        return data;
    }

    public static Map<String, String> duplicateUser() {
        Map<String, String> data = new HashMap<>();
        data.put("name", "Exigent Services Pvt Ltd");
        data.put("email", "exigent@mailinator.com");
        data.put("password", "Test@1234");
        data.put("confirmPassword", "Test@1234");
        return data;
    }

    public static Map<String, String> noName() {
        String password = Utilities.generateRandomPassword();
        Map<String, String> data = new HashMap<>();
        data.put("email", Utilities.getEmail("default"));
        data.put("password", password);
        data.put("confirmPassword", password);
        return data;
    }

    public static Map<String, String> noEmail() {
        String password = Utilities.generateRandomPassword();
        Map<String, String> data = new HashMap<>();
        data.put("name", Utilities.getCompanyName());
        data.put("password", password);
        data.put("confirmPassword", password);
        return data;
    }

    public static Map<String, String> noPassword() {
        String password = Utilities.generateRandomPassword();
        Map<String, String> data = new HashMap<>();
        data.put("name", Utilities.getCompanyName());
        data.put("email", Utilities.getEmail("default"));
        // data.put("password", password);
        data.put("confirmPassword", password);
        return data;
    }

    public static Map<String, String> passwordWithLessCharacters() {
        Map<String, String> data = new HashMap<>();
        data.put("name", Utilities.getCompanyName());
        data.put("email", Utilities.getEmail("default"));
        data.put("password", "Test");
        data.put("confirmPassword", "Test");
        return data;
    }

    public static Map<String, String> noConfirmPassword() {
        String password = Utilities.generateRandomPassword();
        Map<String, String> data = new HashMap<>();
        data.put("name", Utilities.getCompanyName());
        data.put("email", Utilities.getEmail("default"));
        data.put("password", password);
        // data.put("confirmPassword", password);
        return data;
    }

    public static Map<String, String> plainEmail() {
        String password = Utilities.generateRandomPassword();
        Map<String, String> data = new HashMap<>();
        data.put("name", Utilities.getCompanyName());
        data.put("email", Utilities.getEmail("plainEmail"));
        data.put("password", password);
        data.put("confirmPassword", password);
        return data;
    }

    public static Map<String, String> noLocalPart() {
        String password = Utilities.generateRandomPassword();
        Map<String, String> data = new HashMap<>();
        data.put("name", Utilities.getCompanyName());
        data.put("email", Utilities.getEmail("noLocalPart"));
        data.put("password", password);
        data.put("confirmPassword", password);
        return data;
    }

    public static Map<String, String> onlyLocalPart() {
        String password = Utilities.generateRandomPassword();
        Map<String, String> data = new HashMap<>();
        data.put("name", Utilities.getCompanyName());
        data.put("email", Utilities.getEmail("localPartOnly"));
        data.put("password", password);
        data.put("confirmPassword", password);
        return data;
    }

    public static Map<String, String> doubleAt() {
        String password = Utilities.generateRandomPassword();
        Map<String, String> data = new HashMap<>();
        data.put("name", Utilities.getCompanyName());
        data.put("email", Utilities.getEmail("doubleAt"));
        data.put("password", password);
        data.put("confirmPassword", password);
        return data;
    }

    public static Map<String, String> withoutTopLevelDomain() {
        String password = Utilities.generateRandomPassword();
        Map<String, String> data = new HashMap<>();
        data.put("name", Utilities.getCompanyName());
        data.put("email", Utilities.getEmail("withoutTopLevelDomain"));
        data.put("password", password);
        data.put("confirmPassword", password);
        return data;
    }

    public static Map<String, String> tailingWithDot() {
        String password = Utilities.generateRandomPassword();
        Map<String, String> data = new HashMap<>();
        data.put("name", Utilities.getCompanyName());
        data.put("email", Utilities.getEmail("tailingWithDot"));
        data.put("password", password);
        data.put("confirmPassword", password);
        return data;
    }

    public static Map<String, String> topLevelDomainTooShort() {
        String password = Utilities.generateRandomPassword();
        Map<String, String> data = new HashMap<>();
        data.put("name", Utilities.getCompanyName());
        data.put("email", Utilities.getEmail("topLevelDomainTooShort"));
        data.put("password", password);
        data.put("confirmPassword", password);
        return data;
    }

    public static Map<String, String> domainStartsWithHyphen() {
        String password = Utilities.generateRandomPassword();
        Map<String, String> data = new HashMap<>();
        data.put("name", Utilities.getCompanyName());
        data.put("email", Utilities.getEmail("domainStartsWithHyphen"));
        data.put("password", password);
        data.put("confirmPassword", password);
        return data;
    }

    public static Map<String, String> doubleDotInDomain() {
        String password = Utilities.generateRandomPassword();
        Map<String, String> data = new HashMap<>();
        data.put("name", Utilities.getCompanyName());
        data.put("email", Utilities.getEmail("doubleDotInDomain"));
        data.put("password", password);
        data.put("confirmPassword", password);
        return data;
    }

    public static Map<String, String> noAtTheRate() {
        String password = Utilities.generateRandomPassword();
        Map<String, String> data = new HashMap<>();
        data.put("name", Utilities.getCompanyName());
        data.put("email", Utilities.getEmail("noAtTheRate"));
        data.put("password", password);
        data.put("confirmPassword", password);
        return data;
    }

    public static Map<String, String> illegalChars() {
        String password = Utilities.generateRandomPassword();
        Map<String, String> data = new HashMap<>();
        data.put("name", Utilities.getCompanyName());
        data.put("email", Utilities.getEmail("illegalChars"));
        data.put("password", password);
        data.put("confirmPassword", password);
        return data;
    }

    public static Map<String, String> twoConsicutiveDots() {
        String password = Utilities.generateRandomPassword();
        Map<String, String> data = new HashMap<>();
        data.put("name", Utilities.getCompanyName());
        data.put("email", Utilities.getEmail("twoConsicutiveDots"));
        data.put("password", password);
        data.put("confirmPassword", password);
        return data;
    }

    public static Map<String, String> dotAtBeginning() {
        String password = Utilities.generateRandomPassword();
        Map<String, String> data = new HashMap<>();
        data.put("name", Utilities.getCompanyName());
        data.put("email", Utilities.getEmail("dotAtBeginning"));
        data.put("password", password);
        data.put("confirmPassword", password);
        return data;
    }

    public static Map<String, String> withQuotation() {
        String password = Utilities.generateRandomPassword();
        Map<String, String> data = new HashMap<>();
        data.put("name", Utilities.getCompanyName());
        data.put("email", Utilities.getEmail("withQuotation"));
        data.put("password", password);
        data.put("confirmPassword", password);
        return data;
    }

    public static Map<String, String> unclosedQuotation() {
        String password = Utilities.generateRandomPassword();
        Map<String, String> data = new HashMap<>();
        data.put("name", Utilities.getCompanyName());
        data.put("email", Utilities.getEmail("unclosedQuotation"));
        data.put("password", password);
        data.put("confirmPassword", password);
        return data;
    }

    public static Map<String, String> veryLengthyLocal() {
        String password = Utilities.generateRandomPassword();
        Map<String, String> data = new HashMap<>();
        data.put("name", Utilities.getCompanyName());
        data.put("email", Utilities.getEmail("veryLengthyLocal"));
        data.put("password", password);
        data.put("confirmPassword", password);
        return data;
    }

    public static Map<String, String> veryLengthyDomain() {
        String password = Utilities.generateRandomPassword();
        Map<String, String> data = new HashMap<>();
        data.put("name", Utilities.getCompanyName());
        data.put("email", Utilities.getEmail("veryLengthyDomain"));
        data.put("password", password);
        data.put("confirmPassword", password);
        return data;
    }

    public static Map<String, String> veryLengthyTLD() {
        String password = Utilities.generateRandomPassword();
        Map<String, String> data = new HashMap<>();
        data.put("name", Utilities.getCompanyName());
        data.put("email", Utilities.getEmail("veryLenghtyTLD"));
        data.put("password", password);
        data.put("confirmPassword", password);
        return data;
    }

    public static Map<String, String> missingSecondLevelDomain() {
        String password = Utilities.generateRandomPassword();
        Map<String, String> data = new HashMap<>();
        data.put("name", Utilities.getCompanyName());
        data.put("email", Utilities.getEmail("missingSecondLevelDomain"));
        data.put("password", password);
        data.put("confirmPassword", password);
        return data;
    }

    public static Map<String, String> withComma() {
        String password = Utilities.generateRandomPassword();
        Map<String, String> data = new HashMap<>();
        data.put("name", Utilities.getCompanyName());
        data.put("email", Utilities.getEmail("withComma"));
        data.put("password", password);
        data.put("confirmPassword", password);
        return data;
    }

    public static Map<String, String> doubleConsicutiveAt() {
        String password = Utilities.generateRandomPassword();
        Map<String, String> data = new HashMap<>();
        data.put("name", Utilities.getCompanyName());
        data.put("email", Utilities.getEmail("doubleConsicutiveAt"));
        data.put("password", password);
        data.put("confirmPassword", password);
        return data;
    }

    public static Map<String, String> underscoreInTLD() {
        String password = Utilities.generateRandomPassword();
        Map<String, String> data = new HashMap<>();
        data.put("name", Utilities.getCompanyName());
        data.put("email", Utilities.getEmail("underscoreInTLD"));
        data.put("password", password);
        data.put("confirmPassword", password);
        return data;
    }

    public static Map<String, String> dotBetweenName() {
        String password = Utilities.generateRandomPassword();
        Map<String, String> data = new HashMap<>();
        data.put("name", Utilities.getCompanyName());
        data.put("email", Utilities.getEmail("dotBetweenName"));
        data.put("password", password);
        data.put("confirmPassword", password);
        return data;
    }

    public static Map<String, String> spaceBeforeName() {
        String password = Utilities.generateRandomPassword();
        Map<String, String> data = new HashMap<>();
        data.put("name", Utilities.getCompanyName());
        data.put("email", Utilities.getEmail("spaceBeforeName"));
        data.put("password", password);
        data.put("confirmPassword", password);
        return data;
    }

    public static Map<String, String> spaceAfterName() {
        String password = Utilities.generateRandomPassword();
        Map<String, String> data = new HashMap<>();
        data.put("name", Utilities.getCompanyName());
        data.put("email", Utilities.getEmail("spaceAfterName"));
        data.put("password", password);
        data.put("confirmPassword", password);
        return data;
    }

    public static Map<String, String> dotBeforeAt() {
        String password = Utilities.generateRandomPassword();
        Map<String, String> data = new HashMap<>();
        data.put("name", Utilities.getCompanyName());
        data.put("email", Utilities.getEmail("dotBeforeAt"));
        data.put("password", password);
        data.put("confirmPassword", password);
        return data;
    }

    public static Map<String, String> withoutDomain() {
        String password = Utilities.generateRandomPassword();
        Map<String, String> data = new HashMap<>();
        data.put("name", Utilities.getCompanyName());
        data.put("email", Utilities.getEmail("withoutDomain"));
        data.put("password", password);
        data.put("confirmPassword", password);
        return data;
    }
}